src/wp-includes/rest-api/endpoints/class-wp-rest-comments-controller.php

```diff
@@ -251,11 +251,25 @@ public function get_items( $request ) {
 			$prepared_args['date_query'][0]['before'] = $request['before'];
 		}
 
+		// Set modified_before into date query. Date query must be specified as an array of an array.
+		// Guard against using 'before' and 'modified_before' in the same request.
+		if ( isset( $registered['modified_before'], $request['modified_before'] ) && ! isset( $request['after'] ) ) {
```

Comment:
Can we issue a `400` error in this case instead? I think I'd found it unintuitive as a developer if one of them were silently ignored.

The same in the other checks as well.

Reply:
Added to Posts and Comments controller.